airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py

```diff
@@ -269,21 +271,6 @@ def process_status(
                 (pod_name, namespace, TaskInstanceState.FAILED, annotations, resource_version)
             )
         elif status == "Succeeded":
-            # We get multiple events once the pod hits a terminal state, and we only want to
-            # send it along to the scheduler once.
-            # If our event type is DELETED, we have the POD_EXECUTOR_DONE_KEY, or the pod has
-            # a deletion timestamp, we've already seen the initial Succeeded event and sent it
-            # along to the scheduler.
-            if (
```

Comment:
Hmm, I'm not sure we can get rid of this block completely?

Reply:
Why? When the pod is patched, won't we get a second event, now DELETED (note; meaning its not longer being watched, not that the pod is actually deleted).

Reply:
Yeah, found out this is still needed.